src/api/system.c

```diff
@@ -788,10 +790,12 @@ static int f_load_native_plugin(lua_State *L) {
   const char *basename = strrchr(name, '.');
   basename = !basename ? name : basename + 1;
   snprintf(entrypoint_name, sizeof(entrypoint_name), "luaopen_lite_xl_%s", basename);
-  int (*ext_entrypoint) (lua_State *L, void*) = SDL_LoadFunction(library, entrypoint_name);
+  int (*ext_entrypoint) (lua_State *L, void* (*)(const char*));
+  *(void**)(&ext_entrypoint) = SDL_LoadFunction(library, entrypoint_name);
```

Comment:
This logic is needed because its illegal to mix and match function and data pointers.

Since SDL_LoadFunction (and the OS specific interfaces it uses) all returns generic pointers (void*) we need to somehow degrade our ext_entrypoint function pointer to a data point to assign it.

We do this by casting the address of ext_entrypoint to a void** and then dereferencing it so we get a plain void*.

This is technically Undefined Behavior, but the previous code was Illegal and Undefined Behavior so I guess its an improvement

Reply:
Probably unnecessary on any reasonable OS, but I have no issue with it.

Reply:
Here is a copy of the C Standard as defined by ISO for reference https://port70.net/~nsz/c/c11/n1570.html#6.3.2.3p8

> A pointer to a function of one type may be converted to a pointer to a function of another type and back again; the result shall compare equal to the original pointer. If a converted pointer is used to call a function whose type is not compatible with the referenced type, the behavior is undefined.